observation/point_source_scan.py

```diff
@@ -91,43 +91,43 @@ class NoTargetsUpError(Exception):
                     if not opts.quick and not opts.fine:
                         if opts.source_strength == 'strong' or \
-                           (opts.source_strength == 'auto' and target.flux_density(opts.centre_freq) > 10.0):
-                            session.raster_scan(target, num_scans=5, scan_duration=30, scan_extent=6.0,
-                                                scan_spacing=0.25, scan_in_azimuth=not opts.scan_in_elevation,
-                                                projection=opts.projection)
+                                (opts.source_strength == 'auto' and target.flux_density(opts.centre_freq) > 10.0):
+                            raster_params = dict(num_scans=5, scan_duration=30, scan_extent=6.0, scan_spacing=0.25)
                         else:
-                            session.raster_scan(target, num_scans=5, scan_duration=60, scan_extent=4.0,
-                                                scan_spacing=0.25, scan_in_azimuth=not opts.scan_in_elevation,
-                                                projection=opts.projection)
+                            raster_params = dict(num_scans=5, scan_duration=60, scan_extent=4.0, scan_spacing=0.25)
                     else:  # The branch for Quick and Fine scans
                         if opts.quick:
-                            session.raster_scan(target, num_scans=3, scan_duration=15, scan_extent=5.0,
-                                                scan_spacing=0.5, scan_in_azimuth=not opts.scan_in_elevation,
-                                                projection=opts.projection)
+                            raster_params = dict(num_scans=3, scan_duration=15, scan_extent=5.0, scan_spacing=0.5)
                         elif opts.fine:
-                            session.raster_scan(target, num_scans=5, scan_duration=60, scan_extent=1.0,
-                                                scan_spacing=4. / 60., scan_in_azimuth=not opts.scan_in_elevation,
-                                                projection=opts.projection)
+                            raster_params = dict(num_scans=5, scan_duration=60, scan_extent=1.0, scan_spacing=4/60.)
                         else:  # if opts.search_fine:
-                            session.raster_scan(target, num_scans=9, scan_duration=60, scan_extent=2.0,
-                                                scan_spacing=5. / 60., scan_in_azimuth=not opts.scan_in_elevation,
-                                                projection=opts.projection)
-
+                            raster_params = dict(num_scans=9, scan_duration=60, scan_extent=2.0, scan_spacing=5/60.)
+                    
+                    # Confirm that the target will be "up" for the entire duration.
+                    raster_duration = raster_params["num_scans"] * (raster_params["scan_duration"] + 2) # Extra for slew
+                    if not session.target_visible(target, duration=raster_duration):
+                        continue
```

Comment:
In session's `raster_scan` there is already a check like [this](https://github.com/ska-sa/katcorelib/blob/889825eb1f6e9d38dd194fcba28c63e396a69cd0/katcorelib/mkat_session.py#L1258-L1269):

```python
        # Calculate average time that noise diode is operated per scan,
        # to add to scan duration in check below
        nd_time = session.nd_params['on'] + session.nd_params['off']
        nd_time *= scan_duration / max(session.nd_params['period'],
                                       scan_duration)
        nd_time = nd_time if session.nd_params['period'] >= 0 else 0.
        # Check whether target is visible for entire duration of raster scan
        if not session.target_visible(target,
                                      (scan_duration + nd_time) * num_scans):
            user_logger.warning("Skipping raster scan, as target %r will be "
                                "below horizon", target.name)
            return False
```

Doesn't that work?  Maybe this script just needs to check the return value of `session.raster_scan`.

Reply:
Agreed.

Reply:
i agree, will strip this code from my PR.

i also think the doc comment in session.track(), scan() & raster_scan() should explicitly state that this visibility check is performed. @ajoubertza i'll leave that decision and follow-up to you.

Reply:
hang on, i take that back.

to address the original issue #5 one must skip out before the code reaches session.raster_scan(), since otherwise there will still be a compscan with a small number of dumps but without raster data.

Reply:
True. This is an artifact of the way compscan labels work. If we ever had kattelmod-driven scripts, you could chuck out the raster scan without an actual label. Hey, one is allowed to dream...

Reply:
@ludwigschwardt You can use the MT project.

Reply:
@ludwigschwardt you should make your own ticket - i don't particularly care for the task that you have identified, and you are best placed to describe what you think must be done. i am only interested in seeing my change through against the current code base.

Reply:
@ludwigschwardt what's the status? has the `raster_scan` functionality been reworked?.